istioctl/cmd/precheck.go

```diff
@@ -145,6 +153,32 @@ func checkControlPlane(cli kube.ExtendedClient) (diag.Messages, error) {
 	return msgs, nil
 }
 
+func checkIOPProfile(cli kube.ExtendedClient) (diag.Messages, error) {
```

Comment:
do we want a precheck? Or just make a useful error message?

For example

currently:
```
$ istioctl manifest generate --set profile=remote
Error: failed to read profile foo from : open profiles/remote.yaml: file does not exist
```

after:
```
$ istioctl manifest generate --set profile=remote
Error: profile "remote" has been removed and is equivalent to "default"
```

Or just automatically translate remote to default?

Reply:
Don't automatically translate remote to default, because In the not-to-distant future we would like to reuse "remote" to be the "proper" remote profile, which is currently named "external" (which is a little confusing). When we re-introduce the remote profile (maybe in 1.16?) we will deprecate and eventually remove the external profile. Bottom line, is we would like to make sure that nobody continues to use the current incorrect remote profile. The nicer error message would be fine, but we will have to remove it when we reintroduce the proper remote profile.

It's not clear to me what will happen if users upgrade an existing installation that is using the "remote" profile. Isn't precheck the way to warn about the potential problem?

Reply:
It is the proper way to check about complex things, not simple schema validation. We already have deprecation _warnings_ in operator/pkg/apis/istio/v1alpha1/validation/validation.go; for making it a proper error we should just do it directly.

Not everyone will run `precheck`, but most people that do not will run `install` and get a super obscure error and not know what to do, probably.

Reply:
In that case, I agree that it would be best to just handle this by generating a more useful validation error message, and no need for a precheck.

Reply:
remove the current `checkIOPProfile` func?

Reply:
done

Reply:
```
[root@test-build ~]# ./istioctl manifest generate --set profile=remote
Error: profile "remote" has been removed and is equivalent to "default"
```